src/main/java/teammates/ui/template/FeedbackResponseRow.java

```diff
@@ -52,24 +54,42 @@ public FeedbackResponseRow(int fbIndex, int personIndex, String personType,
                 String showGiverNameToString = StringHelper.removeEnclosingSquareBrackets(frc.showGiverNameTo.toString());
                 String recipientName = results.getNameForEmail(response.recipient);
                 String giverEmail = frc.giverEmail;
-                Map<String, String> instructorEmailNameTable = results.instructorEmailNameTable;
+                Map<String, String> commentGiverEmailNameTable = results.commentGiverEmailNameTable;
                 FeedbackResponseCommentRow responseRow = new FeedbackResponseCommentRow(frc,
                         giverEmail, giverName, recipientName, showCommentTo, showGiverNameToString, responseVisibilities,
-                        instructorEmailNameTable, results.getTimeZone());
+                        commentGiverEmailNameTable, results.getTimeZone());
                 String whoCanSeeComment = null;
                 boolean isVisibilityIconShown = false;
                 if (results.feedbackSession.isPublished()) {
-                    boolean isResponseCommentPublicToRecipient = !frc.showCommentTo.isEmpty();
-                    isVisibilityIconShown = isResponseCommentPublicToRecipient;
+                    isVisibilityIconShown = !frc.showCommentTo.isEmpty();
 
                     if (isVisibilityIconShown) {
                         whoCanSeeComment = getTypeOfPeopleCanViewComment(frc, question);
                     }
                 }
                 responseRow.setVisibilityIcon(isVisibilityIconShown, whoCanSeeComment);
-                responseRow.enableEditDelete();
+                if (responseRow.getGiverRole() == null) {
```

Comment:
just a question, when will this happen and why the default giver role is instructor.

Reply:
@LuLechuan This is for backward compatibility. All the comments were by instructors previously.